asapdiscovery-docking/asapdiscovery/docking/docking_v2.py

```diff
@@ -0,0 +1,283 @@
+
+
+class DockingResult(BaseModel):
+    """
+    Schema for a DockingResult, containing both a DockingInputPair used as input to the workflow
+    and a Ligand object containing the docked pose.
+    Also contains the probability and chemgauss4 score of the docked pose.
+    """
+
+    input_pair: DockingInputPair = Field(description="Input pair")
+    posed_ligand: Ligand = Field(description="Posed ligand")
+    probability: Optional[PositiveFloat] = Field(description="Probability")
+    score_type: SCORE_TYPE = Field(description="Docking score type")
+    score: float = Field(description="Docking score")
+    provenance: dict[str, str] = Field(description="Provenance")
+
+    @root_validator
+    @classmethod
+    def smiles_match(cls, values):
+        posed_ligand = values.get("posed_ligand")
+        input_pair = values.get("input_pair")
+        if posed_ligand.smiles != input_pair.ligand.smiles:
+            raise ValueError(
+                "SMILES of ligand and ligand in input docking pair not match"
+            )
+        return values
+
+    def to_posed_oemol(self) -> oechem.OEMol:
+        """
+        Combine the target and ligand into a single oemol
+        """
+        _, prot, _ = split_openeye_design_unit(self.input_pair.complex.target.to_oedu())
+        return combine_protein_ligand(prot, self.posed_ligand.to_oemol())
+
+
+class DockingBase(BaseModel):
+    """
+    Base class for docking.
+    """
+
+    type: Literal["DockingBase"] = "DockingBase"
+
+    @abc.abstractmethod
+    def _dock() -> list[DockingResult]:
+        ...
+
+    def dock(
+        self, inputs: list[DockingInputPair], use_dask: bool = False, dask_client=None
+    ) -> Union[list[dask.delayed], list[DockingResult]]:
+        if use_dask:
+            delayed_outputs = []
+            for inp in inputs:
+                out = dask.delayed(self._dock)(inputs=[inp])
+                delayed_outputs.append(out[0])  # flatten
+            outputs = actualise_dask_delayed_iterable(
+                delayed_outputs, dask_client=dask_client
+            )
+        else:
+            outputs = self._dock(inputs=inputs)
+        return outputs
+
+    @abc.abstractmethod
+    def provenance(self) -> dict[str, str]:
+        ...
+
+
+class POSIT_METHOD(Enum):
+    ALL = oedocking.OEPositMethod_ALL
+    HYBRID = oedocking.OEPositMethod_HYBRID
+    FRED = oedocking.OEPositMethod_FRED
+    MCS = oedocking.OEPositMethod_MCS
+    SHAPEFIT = oedocking.OEPositMethod_SHAPEFIT
+
+    @classmethod
+    def reverse_lookup(cls, value):
+        return cls(value).name
+
+
+class POSIT_RELAX_MODE(Enum):
+    CLASH = oedocking.OEPoseRelaxMode_CLASHED
+    ALL = oedocking.OEPoseRelaxMode_ALL
+    NONE = oedocking.OEPoseRelaxMode_NONE
+
+
+class POSITDocker(DockingBase):
+    """
+    Docker class for POSIT.
+    """
+
+    type: Literal["POSITDocker"] = "POSITDocker"
+    score_type: Literal[SCORE_TYPE.CHEMGAUSS4] = SCORE_TYPE.CHEMGAUSS4
+
+    relax: POSIT_RELAX_MODE = Field(
```

Comment:
Similar to the score does it make sense to separate the relaxation method, I am not familiar with posit so maybe these are very specific to this method and this makes sense here but more generally it might be nice to have other relaxation methods like they hybrid ML/MM stuff we have in FEGrow.

Reply:
Lets push adding relaxation blocks to the to another PR, as adding relaxation as a seprrate step here isn't really possible as baked into OpenEye

Reply:
issue raised #527